### ISSUE TITLE
Parallel PINN Code Issue

### ISSUE THREAD

**Issue by SoumikSarkar830** [SoumikSarkar830/Laser_PINN#3]
[ParallelPINN.pdf](https://github.com/user-attachments/files/20492040/ParallelPINN.pdf)

For the attached simple Parallel PINN problem (Parallel_PINN_4GPUs.ipynb), it shows "One of the differentiated Tensors appears to not have been used in the graph". This indicates that p is not correctly connected to its input tensor X_f_p, even though X_f_p.requires_grad=True and both live on the same device.

**Comment by raghavg97** [comment]
`` p_y = autograd.grad(p, X_f_p.clone().detach().requires_grad_(True).to(device_v), grad_outputs=torch.ones_like(p), create_graph=True)[0][:, 1:2]``

Try removing ``.requires_grad_(True)``, i.e., change this to 

``p_y = autograd.grad(p, X_f_p.clone().detach().to(device_v), grad_outputs=torch.ones_like(p), create_graph=True)[0][:, 1:2]``
